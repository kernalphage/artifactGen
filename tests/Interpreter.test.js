
import ArtifactGenerator from "..";
let {Scanner, Parser, BasicInterpreter} = ArtifactGenerator.Core;


// TODO: folow better Jest practices for mock setup/teardown
const mockMath = Object.create(global.Math);
let seed = 1;
mockMath.random = ()=> {var x = Math.sin(seed++ *641.23523)*1782.12412; return x - Math.floor(x); };
global.Math = mockMath;
  
  test('basic json value', () => {
    let source = `
   [item]
   description = this is a @metalColor ring that is worth @metalValue;
   pi = 3.14;
   randomItem = 1:10;
   metalColor, metalValue = gold, 200 | silver, 100 | copper, 10 ;
  
   [Second]
   value = "special 324";
   value2 = "324"
  `;
    let scan = new Scanner(source);
    let parse = new Parser(scan.tokens);
    let basic = new BasicInterpreter();
    basic.execute(parse.definitions);
    expect(basic.export()).toMatchSnapshot();
  });
  