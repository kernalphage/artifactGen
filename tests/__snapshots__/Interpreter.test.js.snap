// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic json value 1`] = `
"item: 
pi:3.14
randomItem:6
metalColor:gold
metalValue:200
description:this is a gold ring that is worth 200Second: 
value:special 324
value2:324"
`;
